util/util.go

```diff
@@ -86,6 +86,33 @@ func FileMetaEqual(actual data.FileMeta, expected data.FileMeta) error {
 	return nil
 }
 
+func BytesMatchLenAndHashes(fetched []byte, length int64, hashes data.Hashes) error {
+	flen := int64(len(fetched))
+	if length != 0 && flen != length {
+		return ErrWrongLength{length, flen}
+	}
+
+	if len(hashes) != 0 {
```

Comment:
Why do we need the `!= 0` check? (This is probably my Go inexperience showing)

Reply:
That's right, the `len` check can be omitted since it will skip the `for` loop if there are no `hashes` entries 👍

Reply:
Well spotted both, thanks! This was my Go inexperience showing :-) Removed the `len()` check in [39b9b0f](https://github.com/theupdateframework/go-tuf/pull/325/commits/39b9b0f1c495d2be5ff14317f0248dce5327e05d)